packages/stream_chat/lib/src/core/http/system_environment_manager.dart

```diff
@@ -0,0 +1,60 @@
+// ignore_for_file: use_setters_to_change_properties
+
+import 'package:stream_chat/src/system_environment.dart';
+import 'package:stream_chat/version.dart';
+
+/// {@template systemEnvironmentManager}
+/// A manager class to handle the current [SystemEnvironment].
+/// {@endtemplate}
+class SystemEnvironmentManager {
+  /// {@macro systemEnvironmentManager}
+  SystemEnvironmentManager({
+    SystemEnvironment environment = const SystemEnvironment(
+      sdkName: 'stream-chat',
+      sdkIdentifier: 'dart',
+      sdkVersion: PACKAGE_VERSION,
+    ),
+  }) : _environment = environment;
```

Comment:
We don't want to have people change these values. What do you think of creating 2 constructors like this? (simplified)

```suggestion
  SystemEnvironmentManager.llc() : sdkIdentifier = 'dart';
  SystemEnvironmentManager.core() : sdkIdentifier = 'core';
  SystemEnvironmentManager.flutter() : sdkIdentifier = 'flutter';
```

For Core and Flutter I would add an `AppConfig` parameter, which only contains the app specific settings, such as appName and appVersion. We can make more variables required there.

Reply:
I don't think the `llc` should not know about the packages where it's being used.

Reply:
What we can do is make the `sdkName` and `sdkVersion` field private and keep the other open. wdyt?